src/SimpleRADOSStriper.h

```diff
@@ -0,0 +1,137 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License version 2.1, as published by
+ * the Free Software Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef _SIMPLERADOSSTRIPER_H
+#define _SIMPLERADOSSTRIPER_H
+
+#include <string_view>
+#include <thread>
+
+#include "include/buffer.h"
+#include "include/rados/librados.hpp"
+#include "include/uuid.h"
+#include "include/types.h"
+
+#include "common/ceph_time.h"
+#include "common/perf_counters.h"
+
+class SimpleRADOSStriper
+{
+public:
+  using aiocompletionptr = std::unique_ptr<librados::AioCompletion>;
+  using clock = ceph::coarse_mono_clock;
+  using time = ceph::coarse_mono_time;
+
+  struct extent {
+    std::string soid;
+    size_t len;
+    size_t off;
+  };
+
+  static inline const uint64_t object_size = 22; /* power of 2 */
+  static inline const uint64_t min_growth = (1<<27); /* 128 MB */
+  static inline const char XATTR_EXCL[] = "striper.excl";
```

Comment:
if these are not part of `SimpleRADOSStriper` 's public interface, might want to move them to the `private` section, or even move them into an anonymous namespace in the .cc file?